python/composio/tools/local/sqltool/actions/sql_query.py

```diff
@@ -25,37 +26,52 @@ class SqlQueryResponse(BaseModel):
 
-    def execute(self, request: SqlQueryRequest, metadata: Dict) -> SqlQueryResponse:
-        """Execute SQL query"""
-        # Implement logic to process input and return output
-        # Example:
-        # response_data = {"result": "Processed text: " + request.text}
-        # Implement logic to process input and return output
-
-        # Check if the database file exists
-        if not Path(request.connection_string).exists():
-            raise ValueError(
-                f"Error: Database file '{request.connection_string}' does not exist."
-            )
+    def _is_sqlite_connection(self, connection_string: str) -> bool:
+        """Determine if the connection string is for a SQLite database"""
+        return (
+            connection_string.endswith(".db")
+            or connection_string.endswith(".sqlite")
+            or connection_string.endswith(".sqlite3")
+            or connection_string.startswith("sqlite:///")
+        )
 
+    def execute(self, request: SqlQueryRequest, metadata: Dict) -> SqlQueryResponse:
+        """Execute SQL query for either SQLite or remote databases"""
         try:
-            # Use 'with' statement to manage the database connection
-            with sqlite3.connect(request.connection_string) as connection:
-                cursor = connection.cursor()
-                cursor.execute(request.query)
-                response_data = cursor.fetchall()
-                connection.commit()
-
-            # Prepare the response data
-            return SqlQueryResponse(
-                execution_details={"executed": True},
-                response_data=response_data,
-            )
-
-        except sqlite3.Error as e:
-            raise ValueError(f"SQLite error: {str(e)}") from e
+            if self._is_sqlite_connection(request.connection_string):
+                return self._execute_sqlite(request)
+            else:
+                return self._execute_remote(request)
+        except Exception as e:
+            raise ValueError(f"Database error: {str(e)}") from e
+
+    def _execute_sqlite(self, request: SqlQueryRequest) -> SqlQueryResponse:
+        """Execute query for SQLite database"""
+        db_path = request.connection_string.replace("sqlite:///", "")
+        if not Path(db_path).exists():
+            raise ValueError(f"Error: Database file '{db_path}' does not exist.")
+        with sqlite3.connect(db_path) as connection:
+            cursor = connection.cursor()
+            cursor.execute(request.query)
+            response_data = [list(row) for row in cursor.fetchall()]
+            connection.commit()
+        return SqlQueryResponse(
+            execution_details={"executed": True, "type": "sqlite"},
+            response_data=response_data,
+        )
+
+    def _execute_remote(self, request: SqlQueryRequest) -> SqlQueryResponse:
+        """Execute query for remote databases"""
+        engine = sqlalchemy.create_engine(request.connection_string)
```

Comment:
Consider adding connection timeout and pooling configuration for remote database connections. This could help prevent hanging connections and improve performance in production environments.